web/api/v1/api.go

```diff
@@ -469,13 +469,16 @@ func returnAPIError(err error) *apiError {
 		return nil
 	}
 
-	switch err.(type) {
-	case promql.ErrQueryCanceled:
-		return &apiError{errorCanceled, err}
-	case promql.ErrQueryTimeout:
-		return &apiError{errorTimeout, err}
-	case promql.ErrStorage:
-		return &apiError{errorInternal, err}
+	// The error may have been wrapped, so we should also check the parent ones.
```

Comment:
Unwrap feels like we are looking at child errors (just my opinion, not a blocker)

Reply:
I've used the "parent" nomenclature because it's something "happening before". In this sense, parents come first then children.

Reply:
Sounds fine, no issues